lib/Cake/Cache/Engine/RedisEngine.php

```diff
@@ -228,7 +228,7 @@ public function clearGroup($group) {
  * Disconnects from the redis server
  */
 	public function __destruct() {
-		if (!$this->settings['persistent']) {
+		if (empty($this->settings['persistent']) && !is_null($this->_Redis)) {
```

Comment:
```suggestion
		if (empty($this->settings['persistent']) && $this->_Redis !== null) {
```

Using !== is our idiom as it doesn't use a function call.

Reply:
Makes sense. Done!